doc/internals.rst

```diff
@@ -231,3 +231,101 @@ re-open it directly with Zarr:
     zgroup = zarr.open("rasm.zarr")
     print(zgroup.tree())
     dict(zgroup["Tair"].attrs)
+
+
+How to add a new backend
+------------------------
+
+Adding a new backend for read support to Xarray is easy, and you don't need to integrate your code in Xarray.
+All you need to do is to:
+
+- Implement a function that returns an instance :py:class:``Dataset``
+
+- Create a `BackendEntrypoint`` instance with your function as input.
+
+- Declare such instance as external plugin in your setup.py.
```

Comment:
As someone who's been working on opening custom hdf formats in xarray, declaring an external plugin in setup.py doesn't seem _easy_ to me.

Why can't we simply be allowed to add functions to the engine parameter? Example:

```python
from custom_backend import engine

ds = xr.load_dataset(filename, engine=engine)
```

Reply:
I see our point, and I think we can also add an option to open_dataset to take in input the customed backend class.

Reply:
@Illviljan - this is an interesting idea but probably deserves a more detailed discussion. Can you (or @aurghs open a separate issue to discuss)?

Reply:
Done, see #4838.